### PR TITLE
chore: ConnectionProfile de-obsoleting

### PR DESCRIPTION
**Pull request by pkar70**
GitHub Issue (If applicable): #5659

## PR Type

What kind of change does this PR introduce?

- Refactoring (no functional changes, no api changes)

## What is the current behavior?
a) Setting IsWlanConnectionProfile / IsWwanConnectionProfile used API marked as obsolete.
b) GetConnectionCost() uses API available > 16, without checking Android version.

## What is the new behavior?
a) For newer Android (>22), setting these fields use other API.
b) For early Androids, returns NetworkCostType.Unknown (instead of exception).

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWP/Networking/Connectivity/ConnectionProfile.Android.cs [hunk 1]
```diff
@@ -30,21 +30,52 @@ private ConnectionProfile()
...
+					{
+						if (netCaps.HasTransport(Android.Net.TransportType.Wifi))
+						{
+							IsWlanConnectionProfile = true;
+						}
+						if (netCaps.HasTransport(Android.Net.TransportType.Cellular))
+						{
+							IsWwanConnectionProfile = true;
+						}
+
```
Comment (jeromelaban) [inline src/Uno.UWP/Networking/Connectivity/ConnectionProfile.Android.cs:50]:
```suggestion
						else if (netCaps.HasTransport(Android.Net.TransportType.Cellular))
						{
							IsWwanConnectionProfile = true;
						}
						else 
						{
						   IsWwanConnectionProfile = false;
						}


```

Comment (pkar70) [reply src/Uno.UWP/Networking/Connectivity/ConnectionProfile.Android.cs]: Almost :)  (it can be both Wwan and Wlan).
Ok. Done.

**Comment by jeromelaban** [comment]
Thanks for the PR. Don't push empty commits, just do a rebase and squash your commits.

**Comment by pkar70** [comment]
I think this is not from my PR: `Error : Removed method System.Void Windows.UI.Xaml.DependencyObjectStore.SetResourceBinding(Windows.UI.Xaml.DependencyProperty dependencyProperty, System.Object resourceKey, System.Boolean isTheme, System.Object context) not found in ignore set.`

File: src/Uno.UWP/Networking/Connectivity/ConnectionProfile.Android.cs [hunk 2]
```diff
@@ -15,6 +15,8 @@
 using Windows.Extensions;
 using AndroidConnectivityManager = Android.Net.ConnectivityManager;
 
+// empty change - forcing build restart
+
```
Comment (jeromelaban) [inline src/Uno.UWP/Networking/Connectivity/ConnectionProfile.Android.cs:19]: Can you remove this ?

**Review by jeromelaban — APPROVED** [review]
Thanks for your contribution !

**Review by MartinZikmund — APPROVED** [review]

**Pull request merged**